theano/misc/check_multi_gpu.py

```diff
@@ -0,0 +1,50 @@
+#! /usr/bin/env python
```

Comment:
gpu0 and gpu2 will not work, that's normal

But for cuda0 and cuda2 it should work.  The "Invalid device ordinal" comes from the cuda backend.  So this means that there is no device 0 or it is unavailable on the machine you tried this on.  Which machine was that?